lib/rspec/mocks/proxy.rb

```diff
@@ -101,7 +96,7 @@ def replay_received_message_on(expectation, &block)
             next unless expectation.matches?(actual_method_name, *args)
 
             expectation.safe_invoke(nil)
-            block.call(*args, &received_block) if block
+            block&.call(*args, &received_block)
```

Comment:
I think it was clearer previously, just because something is now available doesn't mean we have to use it.